operators/pkg/utils/k8s/client.go

```diff
@@ -142,3 +146,23 @@ func (s StatusWriter) Update(obj runtime.Object) error {
 		return s.StatusWriter.Update(ctx, obj)
 	})
 }
+
+// NewClient initializes and returns a new Kubernetes client.
+func NewClient(cfg *rest.Config) (client.Client, error) {
```

Comment:
A bit confusing that this `NewClient` function does not return a `Client` interface as defined in this file.
Maybe rename it to `NewClientGo` (similar to the pkg name)?